sdk/azcore/runtime/policy_retry_test.go

```diff
@@ -672,6 +672,45 @@ func TestRetryPolicySuccessWithPerTryTimeoutNoRetryWithBodyDownload(t *testing.T
 	require.Equal(t, largeBody, body)
 }
 
+func TestRetryPolicyWithShouldRetryNoRetry(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithStatusCode(http.StatusRequestTimeout))
+
+	pl := exported.NewPipeline(srv, NewRetryPolicy(&policy.RetryOptions{
+		RetryDelay: time.Millisecond,
+		ShouldRetry: func(r *http.Response, err error) bool {
+			return r.StatusCode != http.StatusRequestTimeout
+		},
+	}))
+	req, err := NewRequest(context.Background(), http.MethodGet, srv.URL())
+	require.NoError(t, err)
+	resp, err := pl.Do(req)
+	require.NoError(t, err)
+	require.EqualValues(t, http.StatusRequestTimeout, resp.StatusCode)
+	require.EqualValues(t, 1, srv.Requests())
+}
+
+func TestRetryPolicyWithShouldRetryRetry(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithStatusCode(http.StatusRequestTimeout))
+	srv.AppendResponse()
+
+	pl := exported.NewPipeline(srv, NewRetryPolicy(&policy.RetryOptions{
+		RetryDelay: time.Millisecond,
+		ShouldRetry: func(r *http.Response, err error) bool {
+			return r.StatusCode == http.StatusRequestTimeout
```

Comment:
The policy retries StatusRequestTimeout by default, so I might add a bool to track whether ShouldRetry was called or use a code the policy wouldn't retry.